### PR TITLE
Support Zookeeper ACL

### PR DESCRIPTION
**Pull request by dmsolr**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
#3256 

### New feature or improvement
Support Cluster-Zookeeper-Plugin config ACL.

### CODE REVIEW

File: oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ClusterModuleZookeeperProvider.java [hunk 1]
```diff
@@ -61,7 +79,39 @@ public ClusterModuleZookeeperProvider() {
...
-        client = CuratorFrameworkFactory.newClient(config.getHostPort(), retryPolicy);
+
+        CuratorFrameworkFactory.Builder builder = CuratorFrameworkFactory.builder()
+            .retryPolicy(retryPolicy)
+            .connectString(config.getHostPort());
+        if (config.isEnableACL()) {
+            final List<ACL> acls = Lists.newArrayList();
+
+            String authInfo = config.getAuth();
+            if ("digest".equals(config.getSchema())) {
```
Comment (wu-sheng) [inline oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ClusterModuleZookeeperProvider.java]: I am not familiar w/ ACL, but why is `digest` schema special?

Comment (dmsolr) [reply oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ClusterModuleZookeeperProvider.java]: When used in the ACL the expression will be the username:base64 encoded SHA1 password digest. But when used in auth the expression must not encode.

The other schemas, such as `IP`, it doesn't have this problem.

File: oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ClusterModuleZookeeperProvider.java [hunk 2]
```diff
@@ -61,7 +79,39 @@ public ClusterModuleZookeeperProvider() {
...
+                try {
+                    authInfo = DigestAuthenticationProvider.generateDigest(authInfo);
+                } catch (NoSuchAlgorithmException e) {
+                    logger.error(e.getMessage(), e);
```
[the remark this answers is not in the document]
Comment (dmsolr) [reply oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ClusterModuleZookeeperProvider.java]: Yes, i will update later.

File: oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ClusterModuleZookeeperProvider.java (hunk 1 above)
Comment (wu-sheng) [reply oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ClusterModuleZookeeperProvider.java]: Is this the official recommended way? Look like strange that, a project like Zookeeper providing such strange style of API.

Comment (dmsolr) [reply oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ClusterModuleZookeeperProvider.java]: It's not good. I will try others.

Comment (dmsolr) [reply oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ClusterModuleZookeeperProvider.java]: The Zookeeper Document said:
> digest uses a username:password string to generate MD5 hash which is then used as an ACL ID identity. Authentication is done by sending the username:password in clear text. When used in the ACL the expression will be the username:base64 encoded SHA1 password digest.

Most projects only provided `digest` schema, such as Solr.

**Comment by wu-sheng** [comment]
What is `ip` schema? Just a literal random String? @kezhenxu94

**Comment by kezhenxu94** [comment]
> What is `ip` schema? Just a literal random String? @kezhenxu94

`ip` schema designates which ip (or ip range) can access the nodes, but in my test, if I pre-create the nodes (say `/skywalking/remote`) and set the ACL to `ip` of my machine (in zookeeper side), and set the `enableACL` to `true` and `schema` to `ip` (in SkyWalking side), it failed to connect to zk, after changing the `enableACL` to `false`, it works, which I think is a little strange

**Comment by dmsolr** [comment]
We use Zookeeper ACL that has 2 steps, 
1. set ACL for `zNode`.
2. authorization, like `login`.

`IP` schema, we need not `authorization` step. But I will do those steps in this pr. @kezhenxu94

**Comment by dmsolr** [comment]
I think whether we support `digest` only. Because `digest` is more widely used, and it is more clear.

**Comment by dmsolr** [comment]
How do you think? @wu-sheng @kezhenxu94

**Comment by wu-sheng** [comment]
@dmsolr One mode is good for me. Make sure it is clear at document.

**Pull request merged**